src/FixedDataTableWidthHelper.js

```diff
@@ -14,137 +14,198 @@
+
+/** 
+ * Calculate new width including flexWidth
+ */
+function addFlexWidth(
+  /*number*/ width,
+  /*number*/ columnFlexGrow = 0,
+  /*number*/ singleColumnFlexWidth
+) /*number*/ {
+  const columnFlexWidth = columnFlexGrow * singleColumnFlexWidth;
+  return width + columnFlexWidth;
 }
 
 function distributeFlexWidth(
   /*array*/ columns,
-  /*number*/ flexWidth
+  /*number*/ totalFlexWidth
 ) /*object*/ {
-  if (flexWidth <= 0) {
+  if (totalFlexWidth <= 0) {
     return {
       columns: columns,
       width: getTotalWidth(columns),
     };
   }
-  var remainingFlexGrow = getTotalFlexGrow(columns);
-  var remainingFlexWidth = flexWidth;
-  var newColumns = [];
-  var totalWidth = 0;
-  for (var i = 0; i < columns.length; ++i) {
-    var column = columns[i];
+
+  let totalWidth = 0;
+  const newColumns = columns.map((column) => {
     if (!column.props.flexGrow) {
       totalWidth += column.props.width;
-      newColumns.push(column);
-      continue;
+      return column;
     }
-    var columnFlexWidth = Math.floor(
+    const columnFlexWidth = Math.floor(
       column.props.flexGrow / remainingFlexGrow * remainingFlexWidth
     );
-    var newColumnWidth = Math.floor(column.props.width + columnFlexWidth);
+
+    const newColumnWidth = addFlexWidth(
+      column.props.width,
+      column.props.flexGrow,
+      getSingleColumnFlexWidth(columns, totalFlexWidth)
+    );
     totalWidth += newColumnWidth;
 
     remainingFlexGrow -= column.props.flexGrow;
     remainingFlexWidth -= columnFlexWidth;
 
-    newColumns.push(React.cloneElement(
+    return React.cloneElement(
       column,
-      {width: newColumnWidth}
-    ));
-  }
+      { width: newColumnWidth }
+    );
+  });
 
   return {
     columns: newColumns,
     width: totalWidth,
   };
 }
 
-function adjustColumnGroupWidths(
-  /*array*/ columnGroups,
-  /*number*/ expectedWidth
-) /*object*/ {
-  var allColumns = [];
-  var i;
-  for (i = 0; i < columnGroups.length; ++i) {
-    React.Children.forEach(
-      columnGroups[i].props.children,
-      (column) => {
-        allColumns.push(column);
-      }
-    );
-  }
-  var columnsWidth = getTotalWidth(allColumns);
-  var remainingFlexGrow = getTotalFlexGrow(allColumns);
-  var remainingFlexWidth = Math.max(expectedWidth - columnsWidth, 0);
-
-  var newAllColumns = [];
-  var newColumnGroups = [];
+/**
+ * Build rekursive tree and calculate group values from descendants
+ */
+function iterateNestedColumns(
+  /*array*/ columns,
+  /*number*/ singleColumnFlexWidth,
+  /*function*/ callback = () => { }
+) /*array*/ {
+  return columns.map((column) => {
+    let newProps;
 
-  for (i = 0; i < columnGroups.length; ++i) {
-    var columnGroup = columnGroups[i];
-    var currentColumns = [];
+    if (!column.type.__TableColumn__) {
+      let width = 0;
 
-    React.Children.forEach(
-      columnGroup.props.children,
-      (column) => {
-        currentColumns.push(column);
+      function getWidth(passedWidth, passedFlexGrow) {
```

Comment:
You're right, `get` isn't the right term here since its not returning that value directly. Instead it causes a side effect. I guess `addToCounter` would be more generic and descriptive.